internal/states/statefile/version4.go

```diff
@@ -389,12 +408,17 @@ func writeStateV4(file *File, w io.Writer, enc encryption.StateEncryption) tfdia
 				continue
 			}
 
+			providerConfigValue := ""
+			if rs.ProviderConfig.Provider.Type != "" {
```

Comment:
Maybe add a helper function IsValid() or IsSet() to the provider addr struct?

Reply:
I was too lazy to do it, so I think this is about time :) https://github.com/opentofu/opentofu/pull/1963/commits#:~:text=Use%20new%20IsSet%20function%20to%20check%20if%20a%20provider%20is%20not%20set